backend/api/pretix/types.py

```diff
@@ -92,6 +163,86 @@ def type(self) -> TicketType:
 
         return TicketType.STANDARD
 
+    @classmethod
+    def from_data(
+        cls,
+        data: ItemDict,
+        language: str,
+        categories: Dict[str, CategoryDict],
+        questions: Dict[str, Question],
+        quotas: Optional[Dict[str, QuotaDict]] = None,
+    ):
+        category = _get_category_for_ticket(data, categories)
+        return cls(
+            id=data["id"],
+            language=language,
+            name=_get_by_language(data, "name", language),
+            description=_get_by_language(data, "description", language),
+            tax_rate=data["tax_rate"],
+            active=data["active"],
+            default_price=data["default_price"],
+            available_from=data["available_from"],
+            available_until=data["available_until"],
+            category=_get_by_language(category, "name", language),
+            category_internal_name=category.get("internal_name", None),
+            variations=[
+                ProductVariation.from_data(variation, language)
+                for variation in data.get("variations", [])
+            ],
+            questions=[
+                Question.from_data(question, language)
+                for question in questions
+                if data["id"] in question["items"]
+            ],
+            quantity_left=cls._get_quantity_left(data, quotas),
+        )
+
+    @staticmethod
+    def _get_quantity_left(data, quotas: Optional[Dict[str, QuotaDict]]):
+        if not bool(data["show_quota_left"]):
+            return None
+
+        # For user's tickets we don't need quantity left
+        if not quotas:
+            return None
+
+        # tickets can be in multiple quotas, in that case the one that has the least amount of tickets
+        # should become the source of truth for availability. See:
+        # https://docs.pretix.eu/en/latest/development/concepts.html#quotas
+        return min(
+            quota["available_number"]
+            for quota in quotas.values()
+            if int(data["id"]) in quota["items"]
+        )
+
+
+@strawberry.type
+class PretixOrderPosition:
```

Comment:
oh I like `AttendeeTicket`! I didn't know how to call it and just `Ticket` I thought was a bit confusing because there is already the `PretixTicket` above...